app/controllers/SchemeTransferOutController.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import forms.YesNoPageFormProvider
+
+import javax.inject.Inject
+import models.Mode
+import models.SchemeId.Srn
+import navigation.Navigator
+import pages.SchemeTransferOutPage
+import play.api.data.Form
+import viewmodels.models.YesNoPageViewModel
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.YesNoPageView
+import services.SaveService
+import SchemeTransferOutController._
+import viewmodels.DisplayMessage.Message
+import viewmodels.implicits._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class SchemeTransferOutController @Inject()(
+  override val messagesApi: MessagesApi,
+  saveService: SaveService,
+  navigator: Navigator,
+  identify: IdentifierAction,
+  allowAccess: AllowAccessActionProvider,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  formProvider: YesNoPageFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: YesNoPageView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  private val form = SchemeTransferOutController.form(formProvider)
+
+  def onPageLoad(srn: Srn, mode: Mode): Action[AnyContent] =
+    identify.andThen(allowAccess(srn)).andThen(getData).andThen(requireData) { implicit request =>
```

Comment:
I prefer `identifyAndRequireData(srn)` action as it reads a bit better than all the `andThen`s